hibernate-core/src/main/java/org/hibernate/Query.java

```diff
@@ -64,6 +65,106 @@
 	 */
 	String getQueryString();
 
+	/**
+	 * "QueryOptions" is a better name, I think, than "RowSelection" -> 6.0
+	 *
+	 * @todo 6.0 rename RowSelection to QueryOptions
+	 *
+	 * @return Return the encapsulation of this query's options, which includes access to
+	 * firstRow, maxRows, timeout and fetchSize.   Important because this gives access to
+	 * those values in their Integer form rather than the primitive form (int) required by JPA.
+	 */
+	RowSelection getQueryOptions();
+
```

Comment:
I really didn't want to move ```#getQueryOptions``` from ```org.hibernate.query.Query``` to ```org.hibernate.Query```, but it was the easiest way to get this to work.

Reply:
What about people using Hibernate ORM 5.3.0, will this break their code?
Should we keep both and deprecate one?

Reply:
`org.hibernate.query.Query` inherits from `org.hibernate.Query`. So Gail just moved this method one level up.

It shouldn't be an issue (except for people overriding `org.hibernate.Query` of course but I don't think it's a common use case).

Reply:
ok, thanks for checking!